src/features/distribute/components/Distributions.tsx

```diff
@@ -21,42 +25,44 @@ export function Distributions() {
   const [confirmDistribution, setConfirmDistribution] = useState<
     Distribution[]
   >([]);
-  const poolAmount = usePoolAmount();
-  const token = usePoolToken();
-  const totalTokens = poolAmount.data?.toString() ?? "0";
-  const distributionResult = api.results.distribution.useQuery({
-    totalTokens,
-  });
-
-  if (distributionResult.isPending) {
+
+  const info = useDistributeInfo();
+  const { data, isFetched } = info;
+
+  const payoutEventsByTransaction = data?.payoutEventsByTransaction;
+  const distributionResult = data?.distributionResult.data;
+  const poolAmount = data?.poolAmount;
+  const token = data?.token.data;
+  const explorerUrl = data?.explorerLink;
+
+  const projectIds = distributionResult?.projectIds || [];
+  const totalVotes = distributionResult?.totalVotes || 0;
+  const totalTokens = poolAmount?.toString() ?? "0";
+  const distributions = importedDistribution.length
+    ? importedDistribution.map((d) => ({
+        ...d,
+        amount:
+          (((d.amountPercentage || 0) / 100) * Number(totalTokens)) /
+          10 ** (token?.decimals ?? 1e18),
```

Comment:
do we have an existing calculation for this already? would be great to use the same everywhere.

Reply:
This one was created from @thelostone-mc & @0xKurt.

We can check this out in a different PR.

Reply:
What are we looking to do here @nijoe1 ? maybe i can tweak the distribution logic to do here

Reply:
This is just calculating the amount that each project should receive based on `percentage * totalTokens`

Reply:
gotcha